protocol-units/bridge/service/src/actions.rs

```diff
@@ -83,6 +88,53 @@ where
 			amount,
 		} => {
 			let future = async move {
+
+				// Check whether recipient is a Movement address (32 bytes)
+				if recipient.0.len() == 32 {
+					tracing::info!("Recipient is a Movement address. Creating faucet client...");
+					let faucet_url = Url::parse("https://faucet.testnet.bardock.movementnetwork.xyz")
+					.expect("Invalid faucet URL");
+					let rest_url = Url::parse("https://testnet.bardock.movementnetwork.xyz")
+					.expect("Invalid REST URL");
+		
+					// Create FaucetClient and RestClient
+					let rest_client = RestClient::new(rest_url.clone());
+					let faucet_client = FaucetClient::new(faucet_url, rest_url);
+		
+					let recipient_address: [u8; 32] = recipient.0.clone().try_into()
+					.map_err(|_| ActionExecError(action.clone(), BridgeContractError::SerializationError))?;
+
+					let account_address = AccountAddress::new(recipient_address);
+
+					/// Fetch and log balance before funding
+					/// This should return an error for a new address because funding creates the account
+					match rest_client.get_account_balance(account_address).await {
+						Ok(balance) => tracing::info!("Balance before funding: {:?}", balance),
+						Err(e) => tracing::error!("Failed to retrieve balance before funding: {:?}", e),
+					}
+					
+					// Execute the funding transaction and capture the result
+					match faucet_client
```

Comment:
Instead of match, you should use `.map_err(|err| ActionExecError(action, err))` or something similar. You don't need to log anything because returned error are logged when catched.
Changing all match you make the code simpler and easier to read.
In fact, you can link all the process using .and_then() after each client process.

Reply:
Agree, matching on Result is overly verbose, you see it in some example code but generally there are more concise way to handle result types.

Reply:
@0xmovses @musitdev I cleaned up the code and refactored the fund_recipient function. 

I tried to remove match from the logic and do this instead:

```
pub async fn fund_recipient(recipient: &BridgeAddress<Vec<u8>>) -> Result<(), BridgeContractError> {
	// Parse URLs
	let faucet_url = Url::parse(MOVEMENT_FAUCET_URL)
	.map_err(|_| BridgeContractError::InvalidUrl)?;
	let rest_url = Url::parse(MOVEMENT_RPC_URL)
	.map_err(|_| BridgeContractError::InvalidUrl)?;
	
	// Create clients
	let rest_client = RestClient::new(rest_url.clone());
	let faucet_client = FaucetClient::new(faucet_url, rest_url);

	// Convert recipient to AccountAddress
	let recipient_address: [u8; 32] = recipient.0.clone().try_into()
	.map_err(|_| BridgeContractError::SerializationError)?;
	let account_address = AccountAddress::new(recipient_address);

	// Check balance before funding
	rest_client.get_account_balance(account_address)
	.await
	.map_err(|_| BridgeContractError::AccountBalanceError)?;

	// Execute the funding transaction
	faucet_client.fund(account_address, 100_000_000)
	.await
	.map_err(|_| BridgeContractError::FundingError)?;

	// Check balance after funding
	rest_client.get_account_balance(account_address)
	.await
	.map_err(|_| BridgeContractError::AccountBalanceError)?;

	Ok(())
}
```

and I also tried mapping to `ActionExecError`. What happens in both cases is I get the error bubbling up and causing the bridge transfer to fail:
<img width="831" alt="image" src="https://github.com/user-attachments/assets/4bff3d55-0c87-422d-b601-8401a7c9d360">

When I removed the balance checks (most recent commit) there's no error but I think if the fund step fails the transfer will fail. So it may be better to use the match pattern? 

Also, do you think BridgeContractError is fine, because ActionExecError contains a BridgeContractError anyway, so for this funding helper function it seems fine to just use BridgeContractError?